packages/fxa-payments-server/src/lib/hooks.tsx

```diff
@@ -14,3 +18,23 @@ export function useCheckboxState(defaultState: boolean = false): useCheckboxStat
   const onChanged = useCallback(ev => setState(ev.target.checked), [ setState ]);
   return [ state, onChanged ];
 }
+
+export function useClickOutsideEffect<T>(onClickOutside: Function) {
```

Comment:
This is a little utility hook that bundles up the functionality of detecting a click outside a given element - i.e. so a dialog message pane can be dismissed. Had to give it a generic type `<T>` because Typescript yells about `useRef()` with a plain `HTMLElement`